src/main/java/io/zksync/signer/ZkSigner.java

```diff
@@ -115,6 +116,8 @@ public ChangePubKey signChangePubKey(ChangePubKey changePubKey) {
             outputStream.write(tokenIdToBytes(changePubKey.getFeeToken()));
             outputStream.write(feeToBytes(changePubKey.getFeeInteger()));
             outputStream.write(nonceToBytes(changePubKey.getNonce()));
+            outputStream.write(numberToBytesBE(changePubKey.getTimeRange().getValidFrom(), 8));
```

Comment:
In the JS SDK the timestamps are serialized the following way:

```
export function serializeTimestamp(time: number): Uint8Array {
    if (time < 0) {
        throw new Error('Negative timestamp');
    }
    return ethers.utils.concat([new Uint8Array(4), numberToBytesBE(time, 4)]);
}
```

I am not sure that the expression `numberToBytesBE(time, 8)` in your code is an equivalent one.

Reply:
Actually it's the same :) Just in JS uses integer 32 with leading zeros but in rust implementation it is u64. I've used the rust's one